src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs

```diff
@@ -331,6 +331,11 @@ private void CreateFileStream()
         {
             Dbg.Assert(_path != null, "FileName is mandatory parameter");
 
+            if (_path == null)
+            {
+                throw new InvalidOperationException("FileName is mandatory parameter");
```

Comment:
Since the strings are now user visible, they need to be moved to a resx file. This applies to all the changes that now throw exceptions.

Reply:
Shall we localize the strings? In the case we should review all code base.

Reply:
@iSazonov I don't believe we localize assert strings because they are for debug only usage. Exception strings are handled like any other display string.

Reply:
@dantraMSFT In this case, I think there is no point in transferring these throw debug strings to the resource files.

Reply:
@iSazonov: I believe there is. We are effectively creating new strings that are displayed to the user. The fact that they were previously assert strings doesn't change that, IMHO.

Reply:
@dantraMSFT What is a guideline? Will we mess the strings with others in the same resx files or we should create new resx files for assert strings? And where we should document this? In CONTRIBUTING.md?

Reply:
We just discuss this becouse never do this previously.

Reply:
Moved.

Reply:
@iSazonov: There already is a guideline in docs\dev-process\coding-guidelines.

Resource strings used for errors or UI should be put in resource files (`.resx`) so that they can be localized later.

With few, ahem, exceptions, strings passed as the message to an exception are error strings and should be in the resx file.  The fact that hit used to be a dbg.assert doesn't change this (IMHO).

Reply:
@dantraMSFT Thanks for clarify!